### DISCUSSION TITLE
When I use XML serialization, there will be an extra trailing '>'

### DISCUSSION THREAD

**Discussion by zms9110750** [dotnet/runtime#71709] [Q&A] [answered]
Something like this
```
<?xml version="1.0"?>
<ArchiveFile ...>...</ArchiveFile>>
```
The code I serialize
```
    public static bool SaveT<T>(string path, T t) where T : class
    {

        try
        {
            XmlSerializer xmlSerializer = new XmlSerializer(typeof(T));
            using var str = File.OpenWrite(path);
            xmlSerializer.Serialize(str, t);
            return true;
        }
        catch (Exception e)
        {
            Debug.Log(path + "\n" + e.Message);
            return false;
        }
    }
```

**Comment by zms9110750** [comment]
I use it for archiving in the game of unity.
This is saved by clicking the button manually, so it should not be caused by sudden interruption.
In addition, sometimes it will follow two '>'

**Comment by PathogenDavid** [comment, marked as answer]
This is happening because `File.OpenWrite` does not truncate the file when it opens it. That extra `>` isn't being written, it's left over from the previous save. ([See the remarks section of the documentation for details](https://docs.microsoft.com/en-us/dotnet/api/system.io.file.openwrite?view=net-6.0#remarks))

You need to use `File.Create` instead.

**Reply by DaZombieKiller** [reply]
Alternatively, you can do `stream.SetLength(0)` before serializing, or `stream.SetLength(stream.Position)` after serializing.
